### ISSUE TITLE
unable to compile with json-c-0.11

### ISSUE THREAD

**Issue by blshkv** [gat3way/hashkill#52]
The detailed report is here:
https://code.google.com/p/pentoo/issues/detail?id=196

**Comment by blshkv** [comment]
and an another bug:

I've tried to compile with "--without-json" option:

```
checking if json-c is available... no                                                                                                                
checking for json_tokener_parse in -ljson... no                                                                                                      
checking json/json.h usability... no                                                                                                                 
checking json/json.h presence... no                                                                                                                  
checking for json/json.h... no                                                                                                                       
checking jsonlib in ... failed
```

but it still failed with the same error message:

```
/bin/sh ../libtool --silent   --tag=CC   --mode=link x86_64-pc-linux-gnu-gcc -fPIC -O3  -fomit-frame-pointer -momit-leaf-frame-pointer -Wall -Wno-for
mat -ftree-vectorize -DBINDIR=\"/usr/bin\" -DDATADIR=\"/usr/share\" -pthread -Wno-unused-value -Wno-unused-result -Wno-switch -D_7ZIP_ST  -flto -fwho
le-program -Wno-psabi  -march=core2 -mtune=generic -O2 -pipe -static-libtool-libs -ldl -static -Bstatic -lcrypto  -L/lib  -lrt -lz -lm -ljson -Wl,-O1
 -Wl,--as-needed -Wl,--hash-style=gnu -o hashkill hashkill-ocl-base.o hashkill-base64.o hashkill-plugins.o hashkill-main.o hashkill-loadfiles.o hashk
ill-threads.o hashkill-hashinterface.o hashkill-sessions.o hashkill-md5_sse2.o hashkill-sha1_sse2.o hashkill-md4_sse2.o hashkill-des_sse2.o hashkill-
ocl-threads.o hashkill-des_nonstd_sse2.o hashkill-7zFile.o hashkill-7zStream.o hashkill-LzmaDec.o hashkill-LzmaEnc.o hashkill-Alloc.o hashkill-LzFind
.o hashkill-lzma.o hashkill-cpu-feat.o hashkill-hashgen.o hashkill-hashgen-mangle.o hashkill-ocl-adl.o hashkill-ocl_md5.o hashkill-ocl_sha1.o hashkil
l-ocl_md4.o hashkill-ocl_ntlm.o hashkill-ocl_md5md5.o hashkill-ocl_mysql5.o hashkill-ocl_sha256.o hashkill-ocl_sha512.o hashkill-ocl_lm.o hashkill-oc
l_nsldap.o hashkill-ocl_joomla.o hashkill-ocl_oscommerce.o hashkill-ocl_ipb2.o hashkill-ocl_vbulletin.o hashkill-ocl_smf.o hashkill-ocl_oracle11g.o h
ashkill-ocl_mssql-2000.o hashkill-ocl_mssql-2005.o hashkill-ocl_nsldaps.o hashkill-ocl_osx-old.o hashkill-ocl_osxlion.o hashkill-ocl_desunix.o hashki
ll-ocl_oracle-old.o hashkill-ocl_mscash.o hashkill-ocl_pixmd5.o hashkill-ocl_phpbb3.o hashkill-ocl_wordpress.o hashkill-ocl_md5unix.o hashkill-ocl_sh
a512unix.o hashkill-ocl_mscash2.o hashkill-ocl_apr1.o hashkill-ocl_wpa.o hashkill-ocl_dmg.o hashkill-unrargpu.o hashkill-ocl_rar.o hashkill-ocl_bfuni
x.o hashkill-ocl_drupal7.o hashkill-ocl_zip.o hashkill-ocl_django256.o hashkill-ocl_sha256unix.o hashkill-ocl_o5logon.o hashkill-ocl_mssql-2012.o has
hkill-ocl_msoffice.o hashkill-ocl_luks.o hashkill-ocl_ripemd160.o hashkill-ocl_whirlpool.o hashkill-cpu-serpent.o hashkill-cpu-twofish.o hashkill-ocl
_truecrypt.o hashkill-ocl_lastpass.o hashkill-ocl_keepass.o hashkill-ocl_mozilla.o hashkill-ocl_pwsafe.o hashkill-ocl_keyring.o hashkill-ocl_kwallet.
o hashkill-bad_blowfish.o hashkill-ocl_msoffice-old.o hashkill-ocl_pdf.o hashkill-ocl_sha384.o hashkill-ocl_odf.o hashkill-ocl_grub2.o hashkill-ocl_a
ndroidfde.o hashkill-ocl_androidpin.o hashkill-ocl_osx-ml.o hashkill-ocl_a51.o libtargets.la                                                         
/usr/lib/gcc/x86_64-pc-linux-gnu/4.7.3/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -ljson                                                    
collect2: error: ld returned 1 exit status 
```

**Comment by gat3way** [comment]
This is rather strange. Is it possible that you have the headers installed, but not the library? Still it should disable json stuff properly though.

**Comment by blshkv** [comment]
No, the library is installed properly. The issue seems exactly the same with
https://github.com/zmap/zmap/pull/105
or 
https://github.com/zmap/zmap/issues/104

Please see the suggested patches in there.

**Comment by gat3way** [comment]
Not the same issue here, but it is fixed now with the latest commit. Sorry for the delay.

**Comment by blshkv** [comment]
'4ffd5451f48e0574da53d073839aa3dfa819c349"

nothing is fixed

the error is still the same:
/usr/lib/gcc/x86_64-pc-linux-gnu/4.7.3/../../../../x86_64-pc-linux-gnu/bin/ld: cannot find -ljson
and I can't compile without json (--without-json)

ps. there is an another error, not sure if it's related:

```
/var/tmp/portage/app-crypt/hashkill-0.3.1_p20131231/work/hashkill-0.3.1_p20131231/missing: Unknown `--is-lightweight' option
```

**Comment by gat3way** [comment]
I am a bit confused...do you get the "cannot find -ljson" when you are trying to build --without-json ?

It may turn out we have two separate issues here, but I am not yet sure what happens there...

**Comment by gat3way** [comment]
OK just some clarification...pkgconfig is not used for json-c...instead there is a custom m4 macro. It tests whether we can compile and link against ljson and if so, it sets the LDFLAGS and CFLAGS to include it...so that's strange, it seems the test goes fine, but when hashkill is linked, it fails to find the lib. Something is wrong with that test....but I am unable to reproduce it on my debian box :(

**Comment by blshkv** [comment]
ok, the second issue is splitted into: https://github.com/gat3way/hashkill/issues/53

**Comment by blshkv** [comment]
this bug is fixed together with #53. I'm unable to compile with older version (0.9), but.. oh well. We move forward.
